app/bundles/CoreBundle/Helper/BuilderTokenHelper.php

```diff
@@ -80,20 +112,20 @@ public function getTokens(
             return;
         }
 
-        $repo   = $this->factory->getModel($this->modelName)->getRepository();
+        $repo   = $this->entityManager->getRepository($this->modelName);
```

Comment:
I think this is what is causing the error that I referenced. The `asset` `$modelName` is not a valid entity, and as such we can't get its repository. We need somehow need to calculate the entity name.